### PR TITLE
*: Support AlterTable with Algorithm option and fix a few things reported by goword

### PR DESCRIPTION
**Pull request by shenli**
Improve the compatibility with MySQL. Some software will send alter table statement with `algorithm = xxx`. 
Refer to the MySQL documents for [AlterTable statement](https://dev.mysql.com/doc/refman/5.7/en/alter-table.html).

### CODE REVIEW

**Comment by jackysp** [comment]
LGTM

File: parser/parser.y [hunk 1]
```diff
@@ -2541,7 +2553,7 @@ TiDBKeyword:
 
 NotKeywordToken:
  "ADDDATE" | "BIT_AND" | "BIT_OR" | "BIT_XOR" | "CAST" | "COUNT" | "CURTIME" | "DATE_ADD" | "DATE_SUB" | "EXTRACT" | "GET_FORMAT" | "GROUP_CONCAT" | "MIN" | "MAX" | "NOW" | "POSITION"
-| "SUBDATE" | "SUBSTRING" | "SUM" | "TIMESTAMPADD" | "TIMESTAMPDIFF" | "TRIM"
+| "SUBDATE" | "SUBSTRING" | "SUM" | "TIMESTAMPADD" | "TIMESTAMPDIFF" | "TRIM" | "INPLACE" | "COPY"
```
Comment (winkyao) [inline parser/parser.y]: Please keep the alphabetical order.

File: ddl/ddl_api.go [hunk 2]
```diff
@@ -859,10 +859,10 @@ func hasAutoIncrementColumn(tbInfo *model.TableInfo) bool {
 }
 
 func (d *ddl) AlterTable(ctx sessionctx.Context, ident ast.Ident, specs []*ast.AlterTableSpec) (err error) {
-	// Only handle valid specs, AlterTableLock is ignored.
+	// Only handle valid specs, AlterTableLock/AlterTableAlgorithm are ignored.
 	validSpecs := make([]*ast.AlterTableSpec, 0, len(specs))
 	for _, spec := range specs {
-		if spec.Tp == ast.AlterTableLock {
+		if spec.Tp == ast.AlterTableLock || spec.Tp == ast.AlterTableAlgorithm {
```
Comment (zimulala) [inline ddl/ddl_api.go]: Could you add a test for it?

**Comment by shenli** [comment]
@winkyao @zimulala PTAL

File: ddl/ddl_worker_test.go [hunk 3]
```diff
@@ -449,3 +449,31 @@ func (s *testDDLSuite) TestCancelJob(c *C) {
 	tblInfo = testTableInfo(c, d, "t1", 3)
 	testCreateTable(c, ctx, d, dbInfo, tblInfo)
 }
+
+func (s *testDDLSuite) TestIgnorableSpec(c *C) {
+
```
Comment (zimulala) [inline ddl/ddl_worker_test.go]: Remove this blank line.
Rest LGTM

**Comment by shenli** [comment]
@zimulala PTAL

**Pull request merged**